trunk/src/protocol/srs_protocol_st.hpp

```diff
@@ -109,7 +148,7 @@ extern bool srs_is_never_timeout(srs_utime_t tm);
 
 // The mutex locker.
 #define SrsLocker(instance) \
-    impl__SrsLocker _SRS_free_##instance(&instance)
+    impl__SrsLocker _SRS_free_instance(instance)
```

Comment:
if this case exist:
```
{
    SrsLocker(mutex1);
    SrsLocker(mutex2);
}
```
I mean is the token pasting still needed?
```suggestion
    impl__SrsLocker _SRS_free_##instance(instance)
```

Reply:
If change to:
```
impl__SrsLocker _SRS_free_##instance(instance)
```

You can not use unique ptr like this:
```
SrsLocker(p.get())
```

The only way is to define a variable:
```
srs_mutex_t lock = *p.get();
SrsLocker(lock);
```

I think if we need to support two lockers, we can define a new macro:
```
#define SrsLocker(lock1, lock2) \
    impl__SrsLocker _SRS_free_instance(lock1); \
    impl__SrsLocker _SRS_free_instance(lock2)
```

Currently I don't see any use case for two locks.